python/dcr_template.py

```diff
@@ -19,137 +20,156 @@
 from scipy.ndimage.interpolation import shift as scipy_shift
-import scipy.optimize.nnls as positive_lstsq
+import scipy.optimize.nnls
 
 import lsst.daf.persistence as daf_persistence
 import lsst.afw.image as afwImage
 import lsst.afw.math as afwMath
 import lsst.meas.algorithms as measAlg
+import lsst.pex.exceptions
 import lsst.pex.policy as pexPolicy
 from lsst.sims.photUtils import Bandpass, PhotometricParameters
 from lsst.utils import getPackageDir
+
 from .calc_refractive_index import diff_refraction
 
 __all__ = ["DcrModel", "DcrCorrection"]
 
 lsst_lat = -30.244639
 lsst_lon = -70.749417
 
-x0 = 150
-dx = 65
-y0 = 480
-dy = 70
-
 
 class DcrModel:
     """Lightweight object that contains only the minimum needed to generate DCR-matched template exposures."""
 
     def __init__(self, model_repository=None, band_name='g', debug_mode=False, **kwargs):
+        """Only run when restoring a model or for testing; otherwise superceded by DcrCorrection __init__.
+
+        @param model_repository: path to the repository where the previously-generated DCR model is stored.
+        @param band_name: name of the bandpass-defining filter of the data. Expected values are u,g,r,i,z,y.
+        @param debug_mode: if set, only use a subset of the data for speed (used in _edge_test)
+        """
         self.debug = debug_mode
-        self.load_model(model_repository=model_repository, band_name=band_name, **kwargs)
-        self.use_fft = False
         self.butler = None
+        self.load_model(model_repository=model_repository, band_name=band_name, **kwargs)
 
-    def generate_templates_from_model(self, obsid_range=None, elevation_arr=None, azimuth_arr=None,
-                                      repository=None, output_repository=None, add_noise=False, use_full=True,
-                                      kernel_size=None, **kwargs):
-        """Use the previously generated model and construct a dcr template image."""
-        exposures = []
-        if repository is not None:
-            butler = daf_persistence.Butler(repository)
-            if self.butler is None:
-                self.butler = butler
-        else:
-            butler = self.butler
+    def generate_templates_from_model(self, obsid_range=None, exposures=None, add_noise=False, use_full=True,
+                                      repository=None, output_repository=None, kernel_size=None, **kwargs):
+        """Use the previously generated model and construct a dcr template image.
+
+        @param obsid_range: single, range, or list of observation IDs in repository to create matched
+                            templates for. Ignored if exposures are supplied directly.
+        @param exposures: optional, list of exposure objects that will have matched templates created.
+        @param add_noise: If set to true, add Poisson noise to the template based on the variance.
+        @param use_full: debugging keyword. Set to use the full psf kernel instead of the average.
```

Comment:
I've removed the colon from the parameter definitions throughout, and hopefully clarified my explanation here.